src/Compilers/Lua/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -297,6 +298,14 @@ public static SyntaxToken Token(SyntaxTriviaList leading, SyntaxKind kind, strin
         public static SyntaxToken Literal(string text, double value) =>
             new(InternalSyntax.SyntaxFactory.Literal(ElasticMarker.UnderlyingNode, text, value, ElasticMarker.UnderlyingNode));
 
+        /// <summary>
+        /// Creates a token with kind NumericLiteralToken from the text and corresponding complex value.
+        /// </summary>
+        /// <param name="text">The raw text of the literal.</param>
+        /// <param name="value">The complex value to be represented by the returned token.</param>
+        public static SyntaxToken Literal(string text, Complex value) =>
+            new(InternalSyntax.SyntaxFactory.Literal(ElasticMarker.UnderlyingNode, text, value, ElasticMarker.UnderlyingNode));
+
```

Comment:
We need something here to validate that the `Complex` number doesn't have a [real](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.complex.real?view=net-6.0#system-numerics-complex-real) part